src/main/java/DAO.java

```diff
@@ -0,0 +1,179 @@
+                java.sql.Timestamp creationDate = resultSet.getTimestamp("creation_date");
+                user = new User(userId, name, creationDate);
+            }
+
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+
+        return user;
+    }
+
+    // Fetch a ticket by ID
+    public Ticket getTicketById(int id) {
+        Ticket ticket = null;
+        String sql = "SELECT id, user_id, ticket_type, creation_date FROM \"Ticket\" WHERE id = ?";
+
+        try (Connection conn = getConnection();
+             PreparedStatement pstmt = conn.prepareStatement(sql)) {
+
+            pstmt.setInt(1, id);
+            ResultSet resultSet = pstmt.executeQuery();
+
+            if (resultSet.next()) {
+                int ticketId = resultSet.getInt("id");
+                int userId = resultSet.getInt("user_id");
+                String ticketType = resultSet.getString("ticket_type");
+                java.sql.Timestamp creationDate = resultSet.getTimestamp("creation_date");
+                ticket = new Ticket(ticketId, userId, ticketType, creationDate);
+            }
+
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+
+        return ticket;
+    }
+
+    // Fetch tickets by user ID
+    public List<Ticket> getTicketsByUserId(int userId) {
+        List<Ticket> tickets = new ArrayList<>();
+        String sql = "SELECT id, user_id, ticket_type, creation_date FROM \"Ticket\" WHERE user_id = ?";
+
+        try (Connection conn = getConnection();
+             PreparedStatement pstmt = conn.prepareStatement(sql)) {
+
+            pstmt.setInt(1, userId);
+            ResultSet resultSet = pstmt.executeQuery();
+
+            while (resultSet.next()) {
+                int id = resultSet.getInt("id");
+                String ticketType = resultSet.getString("ticket_type");
+                java.sql.Timestamp creationDate = resultSet.getTimestamp("creation_date");
+                Ticket ticket = new Ticket(id, userId, ticketType, creationDate);
+                tickets.add(ticket);
+            }
+
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+
+        return tickets;
+    }
+
+    // Method to update ticket type
+    public boolean updateTicketType(int id, String newTicketType) {
+        String sql = "UPDATE \"Ticket\" SET ticket_type = ? WHERE id = ?";
+        boolean updated = false;
+
+        try (Connection conn = getConnection();
+             PreparedStatement pstmt = conn.prepareStatement(sql)) {
+
+            pstmt.setString(1, newTicketType);
+            pstmt.setInt(2, id);
+
+            int rowsAffected = pstmt.executeUpdate();
+            updated = rowsAffected > 0;
+
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+
+        return updated;
+    }
+
+    // Method to delete a user and their tickets
+    public boolean deleteUserById(int userId) {
```

Comment:
the logic can be simplified by using cascades (they are added during the table creation)